source/dem/dem.cc

```diff
@@ -748,23 +732,40 @@ DEMSolver<dim>::check_load_balance_dynamic()
   return load_balance_step;
 }
 
+template <int dim>
+inline bool
+DEMSolver<dim>::is_contact_search_step()
+{
+  if (parameters.model_parameters.contact_detection_method ==
+      Parameters::Lagrangian::ModelParameters::ContactDetectionMethod::constant)
+    return check_contact_search_step_constant();
+  else if (parameters.model_parameters.contact_detection_method ==
```

Comment:
Should we be evaluating if the method is constant at all steps? This could be constant expression no?